src/VirtualClient/VirtualClient.Actions/CtsTraffic/CtsTrafficMetricsParser.cs

```diff
@@ -0,0 +1,121 @@
+                    var header = true;
+
+                    while (!reader.EndOfStream)
+                    {
+                        var line = reader.ReadLine();
+                        var values = line.Split(',');
+                        if (header)
+                        {
+                            header = false;
+                        }
+                        else
+                        {
+                            try
+                            {
+                                // bytes/sec that were sent within the TimeSlice period
+                                this.Metrics.Add(new Metric($"SendBps(TimeSlice-{values[0].Trim()})", Convert.ToDouble(values[1]), "B/s", MetricRelativity.Undefined));
+                            }
+                            catch
+                            {
+                                // do nothing as this result file has non-double values.
+                            }
+
+                            try
+                            {
+                                // bytes/sec that were received within the TimeSlice period
+                                this.Metrics.Add(new Metric($"RecvBps(TimeSlice-{values[0].Trim()})", Convert.ToDouble(values[2]), "B/s", MetricRelativity.Undefined));
+                            }
+                            catch
+                            {
+                                // do nothing as this result file has non-double values.
+                            }
+
+                            try
+                            {
+                                // count of established connections transmitting IO pattern data
+                                this.Metrics.Add(new Metric($"InFlight(TimeSlice-{values[0].Trim()})", Convert.ToDouble(values[3]), MetricRelativity.Undefined));
+                            }
+                            catch
+                            {
+                                // do nothing as this result file has non-double values.
+                            }
+
+                            try
+                            {
+                                // cumulative count of successfully completed IO patterns
+                                this.Metrics.Add(new Metric($"Completed(TimeSlice-{values[0].Trim()})", Convert.ToDouble(values[4]), MetricRelativity.HigherIsBetter));
+                            }
+                            catch
+                            {
+                                // do nothing as this result file has non-double values.
+                            }
+
+                            try
+                            {
+                                // cumulative count of failed IO patterns due to Winsock errors
+                                this.Metrics.Add(new Metric($"NetworkError(TimeSlice-{values[0].Trim()})", Convert.ToDouble(values[5]), MetricRelativity.LowerIsBetter));
+                            }
+                            catch
```

Comment:
why are there so much catch blocks? Is the output pretty flaky?

Reply:
Yes, there is high possibility that one of the column is giving non-double value as this workload works like that, still i want to be sure that flow should not go in catch and further columns should be recorded in the telemetery.